src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/ui/pages/page.py

```diff
@@ -52,6 +50,8 @@ def initialization(pathname):
     project = DashClient[{{ cookiecutter.__solution_definition_name }}].get_project(pathname)
 
     if not project.steps.problem_setup_step.project_initialized:
+        long_running = project.steps.problem_setup_step.upload_bulk_files_to_project_directory()
```

Comment:
I'd recommend changing these methods ``upload_bulk_files_to_project_directory`` and ``get_app_metadata`` to synchronous methods as we need them to run one after the other. This would remove the need for adding the wait between method execution

Reply:
Not sure. I use the MethodStatus checks against these to update the progress bar. If we change the type it will break the logic. Unless it is possible to monitor the status of sync method. Tried but came up empty.

Reply:
You should be able to: https://saf.docs.solutions.ansys.com/version/stable/api/solution.html#ansys.saf.glow.solution.StepModel.get_method_state. If it doesn't work, no worries!